src/main/java/net/snowflake/client/jdbc/SFAsyncResultSet.java

```diff
@@ -102,7 +103,16 @@ public QueryStatus getStatus() throws SQLException {
     if (this.queryID == null) {
       throw new SQLException("QueryID unknown");
     }
-    return session.getQueryStatus(this.queryID);
+    if (this.queryHasSucceeded) {
```

Comment:
Implement caching of query success status in SFAsyncResultSet class. Each resultset is tied to 1 query with 1 query ID and query status. Once a query has succeeded, the results should be available and there is no need to re-fetch the query success status. Don't bother with more server-side calls to re-fetch a successful query status.